src/main/java/step2/domain/LottoRank.java

```diff
@@ -0,0 +1,46 @@
+package step2.domain;
+
+import java.util.Map;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public enum LottoRank {
+    ETC(0, 0),
+    FOURTH(3, 5000),
+    THIRD(4, 50000),
+    SECOND(5, 1500000),
+    FIRST(6, 2000000000);
+
+    private static final Map<Long, LottoRank> LOTTO_RANK_MAP = initLottoRankMap();
+
+    private final long hitCount;
+    private final int prizeMoney;
+
+    LottoRank(long hitCount, int prizeMoney) {
+        this.hitCount = hitCount;
+        this.prizeMoney = prizeMoney;
+    }
+
+    private static Map<Long, LottoRank> initLottoRankMap() {
+
+        Map<Long, LottoRank> result = Stream.of(LottoRank.values())
+                .collect(Collectors.toMap(lottoRank -> lottoRank.hitCount, lottoRank -> lottoRank));
+        result.remove(0L);
+        return result;
+    }
+
+    public static LottoRank toRank(Long hitCount) {
+        if (LOTTO_RANK_MAP.containsKey(hitCount)) {
+            return LOTTO_RANK_MAP.get(hitCount);
+        }
+        return LottoRank.ETC;
```

Comment:
```suggestion
        return Arrays.stream(values())
            .filter(ranking -> ranking.hitCount == hitCount)
            .findFirst()
            .orElse(ETC);
```
Stream을 이용하면 LOTTO_RANK_MAP을 사용하지 않아도 LottoRank를 반환 할 수 있어요. :)

Reply:
우선 처음에는 (별로 이점은 없지만) 캐싱해서 사용해두는 것이 좋을 것 같다고 판단했습니다!

그리고나서 고민한 부분은 0, 1, 2 개를 맞춘 경우를 고려하는 것이였는데요!
LOTTO_RANK_MAP 을 사용해서 2,1,개 맞은 경우를 모두 ETC 로 처리하기위해서 구성했습니다.

대안으로,
6등 - 1개, 0원
5등 - 2개, 0원